### ISSUE TITLE
(1,750,000 SC) - Gitcoin Skynet Exclusive

### ISSUE THREAD

**Issue by MSevey** [NebulousLabs/Skynet-Hive#1]
### Prize Bounty
**Grand Prize 1,000,000 SC!!** (est $2,700)  

There will also be three runner-up **250,000 SC** prizes.

On top of all that, all entrants who produce a substantially interesting Skynet application or upgrade will win a consolation prize of **25,000 SC**. 

### Challenge Description
Build an application which focuses on enhancing Skynet as its fundamental purpose. This application should add a useful feature or functionality to the Skynet ecosystem, making it easy or possible to do something that was previously not easy or possible to do.

Taking an existing open source Skynet application and extending it to support new functionality or otherwise significantly improving the user experience is welcome and encouraged.

### Submission Requirements
 - The project must be open source
 - Skynet must be a fundamental element of the application.
 - The app must do something that is useful and interesting.
 - The app will score better if it has greater appeal to a wider audience
 - Demo videos are strongly recommended for complete submissions

How involved or difficult the application is will not be considered. If a very useful and interesting app can be created with very little effort, this app will still qualify for the grand prize and all other prizes.

Example apps and ideas which would qualify for prizes:
- Pastebin but on Skynet: https://siasky.net/CAAVU14pB9GRIqCrejD7rlS27HltGGiiCLICzmrBV0wVtA
- Skynet video playlist creator: https://siasky.net/nAAJroJW0Z7cYcpev1VVjJXXGU1WF8YQ8eUs8dJHTDEXqQ
- Skynet audio playlist creator: https://siasky.net/IAC_-QuVMlNuwI11kMZOz7w5qeYTL8Bsh0FJfIyp_hcX0Q
- Skynet blogging tool: https://github.com/NebulousLabs/skynet-blogger

Other potential ideas:
- A tool to bookmark and share other Skynet applications and files
- A desktop app for browsing Skynet without using a centralized portal
 - Migrating existing apps from https://awesome.ipfs.io/ to Skynet

### Submission Deadline
March 15th Midnight EST

### Judging Criteria
Submissions will be judged on the following:
 - How applicable is the application and how likely is it to be used by others
 - Potential positive impact on the Skynet ecosystem
 - Works on mainnet

### Winner Announcement Date
March 20th 12pm EST

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.001 DAI (0.0 USD @ $1.0/DAI)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/NebulousLabs/Skynet-Hive/1/4058).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/NebulousLabs/Skynet-Hive/1/4058).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $76,228.59 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 1 month, 3 weeks ago.
Please review their action plans below:


**1) [unbilth](https://gitcoin.co/unbilth) has started work.**

I am a software developer interested in SIA blockchain. 

I don't have a team, I will try to create things to make the sia blockchain and skynet more popular and more accessible.
**2) [walidmujahid](https://gitcoin.co/walidmujahid) has started work.**

Hack with Skynet and have fun. Will post plans under the Github issue and stay as transparent and public as possible with my progress and work.
**3) [developerfred](https://gitcoin.co/developerfred) has started work.**

I have been following siacon's work for some time now and would love to build tools on top of the protocol.
**4) [jjchauhan](https://gitcoin.co/jjchauhan) has started work.**

I have started building application using skynet api. Git repository public in next few days
**5) [jjchauhan](https://gitcoin.co/jjchauhan) has started work.**

I have started building application using skynet api. Git repository public in next few days
**6) [loganbek](https://gitcoin.co/loganbek) has started work.**

Hack together something before March 15th Midnight EST.
**7) [ivanmolto](https://gitcoin.co/ivanmolto) has started work.**

Glad to deploy some ideas to Skynet
**8) [kcchouette](https://gitcoin.co/kcchouette) has started work.**

l'll try to learn more about skynet feature and its ecosystem
**9) [dawe35](https://gitcoin.co/dawe35) has started work.**

I'll listen to https://siasky.net/IAA0NuPdVtZs_ittA7QtAPGjrLaznwnbxti_ZnDC_gn9fA
**10) [oussema](https://gitcoin.co/oussema) has started work.**

You will see my final project at the Github repository
**11) [arun0930](https://gitcoin.co/arun0930) has started work.**

Provide an action plan and any initial questions youรา have for this ticket. Include gitcoin team members usernames and project repo url, if you have one.
(Your response will be reflected in a comment on the Github issue.)
**12) [vbstreetz](https://gitcoin.co/vbstreetz) has started work.**

Customize an existing cloud upload utility to use skynet.
Add download capability among other features.
**13) [amritshenava98](https://gitcoin.co/amritshenava98) has started work.**

Will use to the Sia platform to make a secure storage platform. Currently brainstorming on a few ideas.
**14) [amritshenava98](https://gitcoin.co/amritshenava98) has started work.**

Will use to the Sia platform to make a secure storage platform. Currently brainstorming on a few ideas.
**15) [davebhavin](https://gitcoin.co/davebhavin) has started work.**

Trying to be productive during my spring break. Currently learning about the SIA and its Skynet ecosystem.
**16) [leroyshirto](https://gitcoin.co/leroyshirto) has started work.**

Create a Skynet Adapter for The PHP Leagues Flysystem. https://github.com/thephpleague/flysystem
This would allow Laravel apps to use Skynet as object storage (https://laravel.com/docs/7.x/filesystem). 

Work will be done in https://github.com/leroyshirto/flysystem-skynet
**17) [redsolver](https://gitcoin.co/redsolver) has started work.**

Building a mobile iOS/Android App to manage and stream Music and Podcasts from Skynet.
**18) [brianspha](https://gitcoin.co/brianspha) has started work.**

I will provide regular updates as i make progress
**19) [masaun](https://gitcoin.co/masaun) has started work.**

I will build dApp by using Skynet.
**20) [cpurta](https://gitcoin.co/cpurta) has started work.**

I am planning on building a dashboard web app similar to something like Google Drive for user to upload and view files in a single platform. The web app will be built with Vue.js.
**21) [jay-dee7](https://gitcoin.co/jay-dee7) has started work.**

i'll create a daemon service which syncs a directory to skynet using the small skynet-go package (a small backup solution)

here's the repository where i'll host my code

git@github.com:jay-dee7/sia-box.git
**22) [grilledchickenthighs](https://gitcoin.co/grilledchickenthighs) has started work.**

Music streaming app that will be aan enhancement of the skynet audioplayer mixtape thing
**23) [immense055](https://gitcoin.co/immense055) has started work.**

hi all let's doing hack.any new project friendly hack.need group work.friends given information about work.
**24) [n0name-hpi](https://gitcoin.co/n0name-hpi) has started work.**

I will be working on secure sharing of data files.
**25) [duniaman](https://gitcoin.co/duniaman) has started work.**

I will provide regular updates as i make progress
**26) [happyman811](https://gitcoin.co/happyman811) has started work.**

I plan to create a copy & paste app that will enable a user to copy data from one device and will be able to paste it to a separate device. Provided that the devices use Skynet.

For now, the focus will be to create it as a desktop app,

Action steps:

1.  upload data you want to copy to Skynet
2. download data from select URL
3. Perhaps create a GUI to manage clipboard? (maybe)
**27) [markichu](https://gitcoin.co/markichu) has started work.**

I will work to make some useful apps that utilize Sia Skynet.
**28) [isabelgeek](https://gitcoin.co/isabelgeek) has started work.**

I plan to release an awesome app
**29) [t-900-a](https://gitcoin.co/t-900-a) has started work.**

I will be working on two different projects that complement eachother:

One that uploads media files from emby/jellyfin to skynet.
The script should also produce an rss file, so that others can become aware the new content being available.
https://github.com/t-900-a/jellyfin-skynet-uploader

The other will read an rss feed with sialinks, download the file(s), then execute a cli command supplied by the user. I.e. filebot, to process the file into their own media server.
https://github.com/t-900-a/skynet-rss-reader

I don't have a discord account my email is: t-900@tutanota.com
**30) [hdriqi](https://gitcoin.co/hdriqi) has started work.**

I am building Skyshot, a webpage screenshot service. The screenshot will be available via skynet. This can help developers create website preview on their site.

Github repo: https://github.com/hdriqi/skyshot
Demo: https://siasky.net/AAAa5asP43Vgt-v_65VrNOYWpYJwM4Dra7VcuYoNGuPaYQ
**31) [viraja1](https://gitcoin.co/viraja1) has started work.**

Will work on Sky Audio for this hackathon
**32) [geo-gs](https://gitcoin.co/geo-gs) has started work.**

Update the Sawwit DAPP, which is a DAPP that lets users share ideas on a reddit-styled platform with no centralized backend, to be able to:
 * download files hosted on skynet through both the skynet webportal and siad portal
 * display images inline in the dapp that are hosted on skynet through both the skynet webportal and siad portal
 * display text posts in the dapp that are hosted on skynet through both the skynet webportal and siad portal

Download links should be posted to Sawwit as: `sia://[skylink]`
Inline images should be posted to Sawwit as: `!sia://[skylink]`
Inline text should be posted to Sawwit as: `_sia://[skylink]`

Skynet directory structures are unsupported.
**33) [onyb](https://gitcoin.co/onyb) has started work.**

Skywalker  🌌🚶‍♀️

Skynet is great for sharing files, but it is not very convenient to browse directory structures. We intend to solve this problem by building an app to let users walk through full directory trees uploaded to Skynet.
**34) [latenthero](https://gitcoin.co/latenthero) has started work.**

I'll build a telegram bot that will upload to Skynet any content sent to it (messages, photos, videos etc.)
**35) [akbarnafisa](https://gitcoin.co/akbarnafisa) has started work.**

Skyview is a generator for lazy loading responsive image that uploaded to skynet. This service allows developers to create a fast website by optimizing the initial load time and reduce bandwidth.
**36) [androolloyd](https://gitcoin.co/androolloyd) has started work.**

Im going to work with dan on something cool and we're going to submit it for the bounty.
**37) [avalonche](https://gitcoin.co/avalonche) has started work.**

Making a react native video streaming app
**38) [aj07](https://gitcoin.co/aj07) has started work.**

I will do some cool stuff...

I will iterate on idea. Look on stack. Think what can be done.
Buidl mode and post mode.
**39) [javgh](https://gitcoin.co/javgh) has started work.**

Step 1: Submit my Skynet Hackathon project.
Step 2: ???
Step 3: PROFIT
**40) [reznikmm](https://gitcoin.co/reznikmm) has started work.**

do something interesting and useful
**41) [brandonwestcott](https://gitcoin.co/brandonwestcott) has started work.**

dgit.dev combines the power of git, the [Tupelo DLT](https://docs.tupelo.org/) and [Skynet](https://siasky.net/) to create a fully decentralized, shareable git remote of your project.
**42) [kunalkamble](https://gitcoin.co/kunalkamble) has started work.**

Skynet Folder Explorer Web app:
It is an open-source project to explore Skynet folder structure:

Gitcoin Profile: https://gitcoin.co/kunalkamble
Github Link: https://github.com/kunalkamble/instaskynet-webapp

Member(s): Kunal Kamble (https://github.com/kunalkamble)
**43) [bakaoh](https://gitcoin.co/bakaoh) has started work.**

A CLI for that allow uploading a complex webpage with js, css, image... https://github.com/bakaoh/siasky-cli.git
**44) [ariesjia](https://gitcoin.co/ariesjia) has started work.**

make a mini encrypt file share project
**45) [skynethubio](https://gitcoin.co/skynethubio) has started work.**

Skynet AppStore - Register, View details, Search, Update, Delete your Skynet App (Skapp). This application is deployed on Skynet network. Backend API is on centralized cloud.
**46) [zoek1](https://gitcoin.co/zoek1) has started work.**

I'll improve the Pastebin project adding some helpful functionalities

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/NebulousLabs/Skynet-Hive/1/4058).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__Work for 0.001 DAI (0.0 USD @ $1.0/DAI) has been submitted by__:

 1. [@vporton](https://gitcoin.co/vporton)
 2. [@dawe35](https://gitcoin.co/dawe35)
 3. [@vbstreetz](https://gitcoin.co/vbstreetz)
 4. [@ivanmolto](https://gitcoin.co/ivanmolto)
 5. [@hdriqi](https://gitcoin.co/hdriqi)
 6. [@viraja1](https://gitcoin.co/viraja1)
 7. [@geo-gs](https://gitcoin.co/geo-gs)
 8. [@javgh](https://gitcoin.co/javgh)
 9. [@javgh](https://gitcoin.co/javgh)
 10. [@javgh](https://gitcoin.co/javgh)
 11. [@avalonche](https://gitcoin.co/avalonche)
 12. [@tetek](https://gitcoin.co/tetek)
 13. [@tetek](https://gitcoin.co/tetek)
 14. [@aj07](https://gitcoin.co/aj07)
 15. [@akbarnafisa](https://gitcoin.co/akbarnafisa)
 16. [@brandonwestcott](https://gitcoin.co/brandonwestcott)
 17. [@bakaoh](https://gitcoin.co/bakaoh)
 18. [@kunalkamble](https://gitcoin.co/kunalkamble)
 19. [@reznikmm](https://gitcoin.co/reznikmm)
 20. [@kcchouette](https://gitcoin.co/kcchouette)
 21. [@latenthero](https://gitcoin.co/latenthero)
 22. [@latenthero](https://gitcoin.co/latenthero)
 23. [@happyman811](https://gitcoin.co/happyman811)
 24. [@brianspha](https://gitcoin.co/brianspha)
 25. [@oussema](https://gitcoin.co/oussema)
 26. [@ariesjia](https://gitcoin.co/ariesjia)
 27. [@ariesjia](https://gitcoin.co/ariesjia)
 28. [@ariesjia](https://gitcoin.co/ariesjia)
 29. [@ariesjia](https://gitcoin.co/ariesjia)
 30. [@masaun](https://gitcoin.co/masaun)
 31. [@leroyshirto](https://gitcoin.co/leroyshirto)
 32. [@redsolver](https://gitcoin.co/redsolver)
 33. [@redsolver](https://gitcoin.co/redsolver)
 34. [@redsolver](https://gitcoin.co/redsolver)
 35. [@psudoanon](https://gitcoin.co/psudoanon)
 36. [@isabelgeek](https://gitcoin.co/isabelgeek)
 37. [@unbilth](https://gitcoin.co/unbilth)
 38. [@zoek1](https://gitcoin.co/zoek1)
 39. [@zoek1](https://gitcoin.co/zoek1)
 40. [@skynethubio](https://gitcoin.co/skynethubio)
 41. [@t-900-a](https://gitcoin.co/t-900-a)
 42. [@markichu](https://gitcoin.co/markichu)
 43. [@unbilth](https://gitcoin.co/unbilth)
 44. [@geo-gs](https://gitcoin.co/geo-gs)



@connoroday please take a look at the submitted work:
* [PR](https://github.com/geo-gs/sawwit-ui/pull/1) by @geo-gs
* [PR](https://github.com/unbilth/skylin.git) by @unbilth
* [PR](https://siasky.net/AAAJ0r4P1oGtXQbssokjYtIdsw_ttEs2TuXGC5RCdMS1xA) by @Markichu
* [PR](https://github.com/t-900-a/jellyfin-skynet-uploader) by @t-900-a
* [PR](https://github.com/skynethubio/skhub-appstore-ui) by @skynethubio
* [PR](https://github.com/zoek1/gister) by @zoek1
* [PR](https://github.com/zoek1/gister) by @zoek1
* [PR](https://github.com/unbilth/skylin.git) by @unbilth
* [PR](https://github.com/isabelgeek/sudosky-resolver) by @isabelgeek
* [PR](https://github.com/psudoanon/sia-sound) by @psudoanon
* (Link Not Provided) by @redsolver
* [PR](https://github.com/redsolver/lisky_player) by @redsolver
* [PR](https://github.com/redsolver/lisky_player) by @redsolver
* (Link Not Provided) by @leroyshirto
* [PR](https://github.com/masaun/skynet-asset-marketplace) by @masaun
* [PR](https://github.com/ariesjia/sky-share) by @ariesjia
* [PR](https://github.com/ariesjia/sky-share) by @ariesjia
* [PR](https://github.com/ariesjia/sky-share) by @ariesjia
* [PR](https://github.com/ariesjia/sky-share) by @ariesjia
* [PR](https://github.com/oussema/DecentralizedChatting) by @oussema
* [PR](https://github.com/Brianspha/skyTube) by @Brianspha
* [PR](https://github.com/HappyMan811/Copy_Ninja) by @HappyMan811
* [PR](https://github.com/latenthero/skynet-bot) by @latenthero
* [PR](https://github.com/latenthero/skynet-bot) by @latenthero
* [PR](https://gitlab.com/ykshare/ykshare) by @Kcchouette
* [PR](https://github.com/reznikmm/ada-skynet/pull/1) by @reznikmm
* [PR](https://github.com/kunalkamble/skylink) by @kunalkamble
* [PR](https://github.com/bakaoh/siasky-cli) by @bakaoh
* [PR](https://github.com/quorumcontrol/dgit) by @brandonwestcott
* [PR](https://github.com/akbarnafisa/skyview/) by @akbarnafisa
* [PR](https://github.com/aj07/skygames) by @aj07
* [PR](https://github.com/tetek/skynet-ios) by @tetek
* [PR](https://github.com/tetek/skynet-ios) by @tetek
* [PR](https://github.com/avalonche/skypix) by @avalonche
* [PR](https://github.com/javgh/skygaze) by @javgh
* (Link Not Provided) by @javgh
* [PR](https://github.com/javgh/skygaze) by @javgh
* [PR](https://github.com/geo-gs/sawwit-ui/pull/1) by @geo-gs
* [PR](https://github.com/viraja1/sky_audio) by @viraja1
* [PR](https://github.com/hdriqi/skyshot) by @hdriqi
* [PR](https://github.com/ivanmolto/skynet-air-2048) by @ivanmolto
* [PR](https://github.com/vbstreetz/skynet-cli) by @vbstreetz
* [PR](https://github.com/DaWe35/Skynet-portal-switcher) by @DaWe35
* [PR](https://github.com/vporton/zip-browser4) by @vporton

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/NebulousLabs/Skynet-Hive/1/4058)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/NebulousLabs/Skynet-Hive/1/4058).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $138,532.70 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by vporton** [comment]
It's impossible to build something serious on SkyNet, because:
```
$ curl --head https://siasky.net/CAAVU14pB9GRIqCrejD7rlS27HltGGiiCLICzmrBV0wVtA
HTTP/2 405 
server: nginx/1.14.2
date: Sun, 01 Mar 2020 23:08:42 GMT
content-type: text/plain; charset=utf-8
content-length: 19
allow: GET, OPTIONS
x-content-type-options: nosniff
```
returns a wrong `content-length:`.

**Comment by DaWe35** [comment]
@vporton 
I ran into it yesterday. I wanted to check is the given Skynet hash valid, but when I curl for ONLY HEADER, it returns 405 status. Fun fact, 404 and 405 are not the same, so I can check the difference LOL.

But for more info... You get 405, that's why `content-length = 19`. I think it will be fixed very soon, I submitted an issue here: https://github.com/NebulousLabs/skynet-webportal/issues/58

ScottG showed this trick:
`curl https://siasky.net/[skylink] -i -r 0-0`
